### PR TITLE
simplify Actions config with setup-java@v3 caching support

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

**Comment by SethTisue** [comment]
@griggt you were the one who added the `key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}` stuff. I'm not sure what the motivation was for that, and I don't recall seeing it used in any other repos. do you think we still need it?

**Comment by griggt** [comment]
It seems that `setup-java` now does the right thing automatically for caching (based on the readme, I haven't tried the upgrade myself yet).

An excerpt from the [readme](https://github.com/actions/setup-java#caching-packages-dependencies) (emphasis mine):

> The action has a built-in functionality for caching and restoring dependencies. It uses [actions/cache](https://github.com/actions/cache) under hood for caching dependencies but requires less configuration settings. Supported package managers are gradle, maven and sbt. **The format of the used cache key is `setup-java-${{ platform }}-${{ packageManager }}-${{ fileHash }}`**, where the hash is based on the following files:

    gradle: **/*.gradle*, **/gradle-wrapper.properties
    maven: **/pom.xml
    sbt: all sbt build definition files **/*.sbt, **/project/build.properties, **/project/**.{scala,sbt}

TLDR; looks like we don't need the manual key setting any longer.

**Review by julienrf — APPROVED** [review]
